### PR TITLE
Make BranchSource.strategy use @DataBoundSetter

### PR DESCRIPTION
**Pull request by jglick**
@reviewbybees esp. @stephenc

### CODE REVIEW

File: src/main/java/jenkins/branch/BranchSource.java [hunk 1]
```diff
@@ -78,7 +77,12 @@ public SCMSource getSource() {
      * @return the strategy.
      */
     public BranchPropertyStrategy getStrategy() {
-        return strategy;
+        return strategy != null ? strategy : new DefaultBranchPropertyStrategy(new BranchProperty[0]);
```
Comment (andresrc) [inline src/main/java/jenkins/branch/BranchSource.java]: As `new DefaultBranchPropertyStrategy(new BranchProperty[0])` is effectively constant, couldn't we make it to `private static final` and use it as a default value?

**Comment by andresrc** [comment]
:bee:

**Comment by stephenc** [comment]
Not happy about removing the immutability of this class, but I understand the need :bee:

File: src/main/java/jenkins/branch/BranchSource.java (hunk 1 above)
Comment (jglick) [inline src/main/java/jenkins/branch/BranchSource.java]: Could. The number of instances in the VM is probably only a handful so I saw no need to bother.

**Comment by jglick** [comment]
> Not happy about removing the immutability of this class

I suppose in the future we could define a `@DataBoundBuilder`, if there were some standard implementation of the builder pattern to attach it to.

**Pull request merged**